docs/content/release-notes/flink-1.15.md

```diff
@@ -0,0 +1,597 @@
+It is included in the Flink distribution under `lib/`. For backwards compatibility, users can still swap it with 
+`flink-table-planner_2.12` located in `opt/`. As a consequence, `flink-table-uber` has been split into `flink-table-api-java-uber`, 
+`flink-table-planner(-loader)`, and `flink-table-runtime`. `flink-sql-client` has no Scala suffix anymore.
+
+It is recommended to let new projects depend on `flink-table-planner-loader` (without Scala suffix) in provided scope.
+
+Note that the distribution does not include the Scala API by default.
+Scala users need to explicitly add a dependency to `flink-table-api-scala` or `flink-table-api-scala-bridge`.
+
+#### Remove flink-scala dependency from flink-table-runtime 
+
+##### [FLINK-25114](https://issues.apache.org/jira/browse/FLINK-25114)
+
+The `flink-table-runtime` has no Scala suffix anymore.
+Make sure to include `flink-scala` if the legacy type system (based on TypeInformation) with case classes is still used within Table API.
+
+#### flink-table uber jar should not include flink-connector-files dependency 
+
+##### [FLINK-24687](https://issues.apache.org/jira/browse/FLINK-24687)
+
+The table file system connector is not part of the `flink-table-uber` JAR anymore but is a dedicated (but removable)
+`flink-connector-files` JAR in the `/lib` directory of a Flink distribution.
+
+## JDK Upgrade
+
+The support of Java 8 is now deprecated and will be removed in a future release 
+([FLINK-25247](https://issues.apache.org/jira/browse/FLINK-25247)). We recommend 
+all users to migrate to Java 11.
+
+The default Java version in the Flink docker images is now Java 11 
+([FLINK-25251](https://issues.apache.org/jira/browse/FLINK-25251)). 
+There are images built with Java 8, tagged with “java8”. 
+
+## Drop support for Scala 2.11
+
+Support for Scala 2.11 has been removed in
+[FLINK-20845](https://issues.apache.org/jira/browse/FLINK-20845). 
+All Flink dependencies that (transitively)
+depend on Scala are suffixed with the Scala version that they are built for, for
+example `flink-streaming-scala_2.12`. Users should update all Flink dependecies,
+changing "2.11" to "2.12".
+
+Scala versions (2.11, 2.12, etc.) are not binary compatible with one another. That
+also means that there's no guarantee that you can restore from a savepoint, made
+with a Flink Scala 2.11 application, if you're upgrading to a Flink Scala 2.12
+application. This depends on the data types that you have been using in your
+application.
+
+Also, the Scala Shell/REPL has been removed in
+[FLINK-24360](https://issues.apache.org/jira/browse/FLINK-24360).
+
+## DataStream API
+
+#### TypeSerializer version mismatch during eagerly restore 
+
+##### [FLINK-24858](https://issues.apache.org/jira/browse/FLINK-24858)
+
+This ticket resolves an issue that during state migration between Flink versions
+the wrong serializer might have been picked.
```

Comment:
Is this relevant for the upgrade? Sounds more like a plain bug fix that users don't need to be aware of.

Reply:
Yes, it should be a plain bug fix. I'll then remove it.